### PR TITLE
Disable split train/validation dataset for the unsupervised learning model

### LINKED ISSUE #875: [unsupervised learning]Don't need to split train/evaluate dataset if no label clause
For most cases of the unsupervised learning models, users don't need to specify `LABEL clause` in SQLFlow, sqlflow server also doesn't need to split train/validation dataset.

### PR DESCRIPTION
**Pull request by Yancey0623**
fixed #875

### CODE REVIEW

File: sql/executor.go [hunk 1]
```diff
@@ -282,6 +282,14 @@ func runExec(slct string, db *DB) *PipeReader {
 	return rd
 }
 
+func isUnsupervisedLearning(pr *extendedSelect) bool {
```
Comment (weiguoz) [inline sql/executor.go:285]: How about adding a unit test?

**Review by typhoonzero — APPROVED** [review]
LGTM++

File: sql/executor.go [hunk 2]
```diff
@@ -313,13 +321,14 @@ func runExtendedSQL(slct string, db *DB, modelDir string, session *pb.Session) *
 				if os.Getenv("SQLFLOW_submitter") == "elasticdl" {
 					return elasticDLTrain(wr, pr, db, cwd, session, nil)
 				}
-				// TODO(weiguo): fix the hard code 0.8
-				ds, e := newTrainAndValDataset(db, pr.standardSelect.String(), pr.standardSelect.tables[0], 0.8)
-				if e != nil {
-					return e
+				var ds *trainAndValDataset
+				if !isUnsupervisedLearning(pr) {
```
Comment (BlackPoint-CX) [inline sql/executor.go:325]: If we use `isUnsupervisedLearning` to skip `newTrainAndValDataset` process, how can we make sure the logic of checking  existing table with same name will be run when invoke `createDataset` in `newTrainAndValDataset` ?

File: sql/executor.go (hunk 1 above)
Comment (wangkuiyi) [reply sql/executor.go:285]: I agree to add one, preferably in a followup pull request.

**Pull request merged**